content/en/docs/languages/ruby/basics.md

```diff
@@ -334,6 +334,31 @@ resps.each do |r|
 end
 ```
 
+Streaming RPCs block until the server closes the connection. For non-blocking
+operation, multiple threads must be utilizes as well as the `return_op: true`
+flag. The flag changes the response of the call from a blocking enumerable to a
+`Operation` which gives some control over the connection. The blocking enumerable
+may be processed in a separate thread. This is useful for connecting to persistent
+RPC sessions that keeps the connection open indefinitely.
+
+```ruby
+op = stub.list_features(LIST_FEATURES_RECT, return_op: true)
+Thread.new do 
+  resps = op.execute
+  resps.each do |r|
+    p "- found '#{r.name}' at #{r.location.inspect}"
+  end
+rescue GRPC::Cancelled => e
+  p "operation cancel called - #{e}"
+end
+
+# controls for the operation
+op.status
+op.cancelled?
+op.cancel # terminates connection and raises GRPC::Cancelled in the thread.
```

Comment:
nit: this is subtle, but calling `cancel` doesn't actually guarantee that the call will terminate with `GRPC::Cancelled`.

In general, there is a fundamental race condition in grpc, between cancelling an RPC and that same RPC terminating for some other reason.

So, if we call cancel, and at the same time, an OK status is received on the wire, the call might actually complete successfully.